### ISSUE TITLE
pytorch 1.11 cross entropy loss returns nan with ignore index labels

### ISSUE THREAD

**Issue by HKEa** [pytorch/pytorch#75181]
### 🐛 Describe the bug

import torch
import torch.nn as nn

t=torch.tensor([[1,2,3.]])
l=torch.tensor([-1])
loss=nn.CrossEntropyLoss(ignore_index=-1)
loss(t,l)

### Versions

NGC  nvcr.io/nvidia/pytorch:22.02-py3 image

PyTorch version: 1.11.0a0+17540c5
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.21.3
Libc version: glibc-2.31

Python version: 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:59:51)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-97-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.6.55
GPU models and configuration:
GPU 0: NVIDIA A100-SXM4-40GB
GPU 1: NVIDIA A100-SXM4-40GB
GPU 2: NVIDIA A100-SXM4-40GB
GPU 3: NVIDIA A100-SXM4-40GB
GPU 4: NVIDIA A100-SXM4-40GB
GPU 5: NVIDIA A100-SXM4-40GB
GPU 6: NVIDIA A100-SXM4-40GB
GPU 7: NVIDIA A100-SXM4-40GB

Nvidia driver version: 470.82.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.3.2
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.3.2
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.2
[pip3] pytorch-quantization==2.1.2
[pip3] torch==1.11.0a0+17540c5
[pip3] torch-tensorrt==1.1.0a0
[pip3] torchtext==0.12.0a0
[pip3] torchvision==0.12.0a0
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.5                      281    conda-forge
[conda] mkl-include               2019.5                      281    conda-forge
[conda] mypy_extensions           0.4.3            py38h578d9bd_4    conda-forge
[conda] numpy                     1.22.2           py38h6ae9a64_0    conda-forge
[conda] pytorch-quantization      2.1.2                    pypi_0    pypi
[conda] torch                     1.11.0a0+17540c5          pypi_0    pypi
[conda] torch-tensorrt            1.1.0a0                  pypi_0    pypi
[conda] torchtext                 0.12.0a0                 pypi_0    pypi
[conda] torchvision               0.12.0a0                 pypi_0    pypi

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by cpuhrsch** [comment]
@HKEa - thank you for submitting the issue! Could I also ask you what the expected behavior is?

**Comment by HKEa** [comment]
Prior to 1.11 the returned loss is 0

**Comment by cpuhrsch** [comment]
@HKEa - Makes sense and I'm also able to reproduce this. Could I also ask you why you need this behavior? In both cases all targets have been ignored.

**Comment by HKEa** [comment]
Yes, let say this CE loss is only one of many, instead of not changing the
general loss, it turns it to nan, even if autograd behaviour doesn't change
it mess up the logs
Thank you for the brief reply
Appreciate it.
Haim

**Comment by cpuhrsch** [comment]
@HKEa - I see, so you're logging a running average (or such) of your loss and every once in a while end up with a datapoint without any target. Because CE loss returns `nan` now it screws up that running average whereas previously it just added 0 to it?

**Comment by HKEa** [comment]
Yes, (of course it can be circumvented by some condition)

**Comment by cpuhrsch** [comment]
@HKEa - actually after bisecting the last ~500 commits I found the PR that changed the behavior and an existing issue https://github.com/pytorch/pytorch/issues/70348 that discusses this. I'm going to close this issue just to deduplicate, but could I kindly encourage you to comment on https://github.com/pytorch/pytorch/issues/70348 and state your concern there too?

**Comment by HKEa** [comment]
Yes
